x-pack/plugins/cloud/public/components/chat/iframe.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useRef, useState, CSSProperties } from 'react';
+import { css } from '@emotion/react';
+import { useChat } from '../../services';
+
+type UseChatType =
+  | { enabled: false }
+  | {
+      enabled: true;
+      src: string;
+      ref: React.MutableRefObject<HTMLIFrameElement | null>;
+      style: CSSProperties;
+    };
+
+const MESSAGE_READY = 'driftIframeReady';
+const MESSAGE_RESIZE = 'driftIframeResize';
+const MESSAGE_SET_CONTEXT = 'driftSetContext';
+
+const iframeStyle = css`
```

Comment:
Can chat be hidden from the settings? Do we need to account for that here?

Reply:
Not at present, no.  It's a good idea for a follow-up, both from a Kibana perspective, but also force the IFRAME to provide these styles, (line 104)